### PR TITLE
koord-manager: add webhook to check slo-config

### PR DESCRIPTION
**Pull request by chzhj**
1. check config if valid
2. detect config conflict by node

### Ⅰ. Describe what this PR does
see details https://github.com/koordinator-sh/koordinator/issues/1204

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -36,6 +36,12 @@ const (
 	// ElasticQuotaValidatingWebhook enables validating webhook for ElasticQuotas creations or updates
 	ElasticQuotaValidatingWebhook featuregate.Feature = "ElasticValidatingWebhook"
 
+	//NodeValidatingWebhook enables validating webhook for Node Creation or updates
+	NodeValidatingWebhook featuregate.Feature = "NodeValidatingWebhook"
```
Comment (saintube) [inline pkg/features/features.go:40]: Why does the webhook reject the node create/update when the slo-config conflicts?

Comment (saintube) [reply pkg/features/features.go:40]: How about recording events when the slo-config duplicates on a node instead? The duplication of the slo-config may not be a fatal error that can block the node creation.

Comment (saintube) [reply pkg/features/features.go:40]: okay

**Comment by hormes** [comment]
The code volume of this PR is a bit large, but the unittest coverage is done very well.

**Comment by zwzhang0107** [comment]
/lgtm
However, maybe we can find some better way to do the validation, instead of checking fields one by one.
For example, go-playground/validator provides annotation on struct fields for min/max/format checking.

**Comment by chzhj** [comment]
> /lgtm However, maybe we can find some better way to do the validation, instead of checking fields one by one. For example, go-playground/validator provides annotation on struct fields for min/max/format checking.



> /lgtm However, maybe we can find some better way to do the validation, instead of checking fields one by one. For example, go-playground/validator provides annotation on struct fields for min/max/format checking.

**Comment by chzhj** [comment]
> /lgtm However, maybe we can find some better way to do the validation, instead of checking fields one by one. For example, go-playground/validator provides annotation on struct fields for min/max/format checking.

This is a good suggestion, but the API needs to be modified, I can create a new PR to refactor this

File: pkg/webhook/cm/plugins/sloconfig/selector.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sloconfig
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type virtualNode = map[string]string
```
Comment (eahydra) [inline pkg/webhook/cm/plugins/sloconfig/selector.go:21]: what's the virtualNode?

File: pkg/webhook/cm/plugins/sloconfig/slo_config_plugin.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	"k8s.io/klog/v2"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrladmission "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/koordinator-sh/koordinator/pkg/util/parallelize"
+	"github.com/koordinator-sh/koordinator/pkg/util/sloconfig"
+)
+
+const (
+	PluginName = "SLOControllerConfig"
```
Comment (eahydra) [inline pkg/webhook/cm/plugins/sloconfig/slo_config_plugin.go:35]: It is better to name it `SLOConfig`, consistent with the package name

File: config/rbac/role.yaml [hunk 4]
```diff
@@ -5,6 +5,22 @@ metadata:
   creationTimestamp: null
   name: manager-role
 rules:
+- apiGroups:
+  - ""
+  resources:
```
[the remark this answers is not in the document]
Comment (chzhj) [reply config/rbac/role.yaml]: done

File: config/webhook/manifests.yaml [hunk 5]
```diff
@@ -52,6 +52,27 @@ metadata:
...
+- admissionReviewVersions:
+  - v1beta1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /validate-configmap
+  failurePolicy: Fail
+  name: vconfigmap.kb.io
+  rules:
```
[the remark this answers is not in the document]
Comment (chzhj) [reply config/webhook/manifests.yaml:70]: done

File: pkg/features/features.go [hunk 6]
```diff
@@ -36,6 +36,12 @@ const (
 	// ElasticQuotaValidatingWebhook enables validating webhook for ElasticQuotas creations or updates
 	ElasticQuotaValidatingWebhook featuregate.Feature = "ElasticValidatingWebhook"
 
+	//NodeValidatingWebhook enables validating webhook for Node Creation or updates
```
[the remark this answers is not in the document]
Comment (chzhj) [reply pkg/features/features.go]: done

File: pkg/features/features.go [hunk 7]
```diff
@@ -36,6 +36,12 @@ const (
...
+	NodeValidatingWebhook featuregate.Feature = "NodeValidatingWebhook"
+
+	//ConfigMapValidatingWebhook enables validating webhook for configmap Creation or updates
```
[the remark this answers is not in the document]
Comment (chzhj) [reply pkg/features/features.go]: done

File: pkg/util/parallelize/parallelism.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2022 The Koordinator Authors.
```
[the remark this answers is not in the document]
Comment (chzhj) [reply pkg/util/parallelize/parallelism.go:2]: done

File: pkg/webhook/cm/plugins/sloconfig/selector.go (hunk 2 above)
Comment (chzhj) [reply pkg/webhook/cm/plugins/sloconfig/selector.go]: add comments done

File: pkg/webhook/cm/plugins/sloconfig/slo_config_plugin.go (hunk 3 above)
Comment (chzhj) [reply pkg/webhook/cm/plugins/sloconfig/slo_config_plugin.go]: done

**Review by eahydra — APPROVED** [review]
/lgtm
/approve

**Review by hormes — APPROVED** [review]

**Pull request merged**